src/Giraffe/HttpHandlers.fs

```diff
@@ -351,29 +351,27 @@ let renderHtml (document : XmlNode) : HttpHandler =
 /// Example:
 /// let negotiationRules = dict [ "application/json", json; "application/xml" , xml ]
 /// `json` and `xml` are both the respective default HttpHandler functions in this example.
-let negotiateWith (negotiationRules    : IDictionary<string, obj -> HttpHandler>)
+let negotiateWith (negotiationRules    : #IDictionary<string, obj -> HttpHandler>)
                   (unacceptableHandler : HttpHandler)
                   (responseObj         : obj)
                   : HttpHandler =
     fun (next : HttpFunc) (ctx : HttpContext) ->
         (ctx.Request.GetTypedHeaders()).Accept
         |> fun acceptedMimeTypes ->
             match isNull acceptedMimeTypes || acceptedMimeTypes.Count = 0 with
-            | true  ->
-                negotiationRules.Keys
+            | true ->
+                negotiationRules
                 |> Seq.head
```

Comment:
Not a reflection on your tidy but of original code, I think this is a bug waiting to happen as IDictionary is an interface on any Map/Dictionary and worked to date given we are using Generic Dictionary that happens to enumerate in order added, if a Map was passed, the first value would be different. I think it makes more sense to have default handler as an argument to function rather then a magic entry in a dictionary that feels quite hacked and wont work properly in many cases

Reply:
you can't pass a map.

Reply:
But Map has an IDictionary Interface implemented, what stops function accepting it?

Reply:
in any case the fallback needs to be made explicit